stock_quant_sale_order/wizard/stock_quant_sale_order.py

```diff
@@ -0,0 +1,110 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields, api, _
+from odoo.exceptions import UserError
+
+
+class QuantSaleOrderWizard(models.TransientModel):
+    _name = 'quant.sale.order.wizard'
+
+    partner_id = fields.Many2one(
+        'res.partner',
+        string='Customer',
+        required=True,
+    )
+    team_id = fields.Many2one(
+        'crm.team',
+        string='Sales Channel',
+        required=True,
+    )
+
+    @api.model
+    def default_get(self, fields):
+        """Raise to warning in
+            - selected quants not in same location.
+            - selected quants are already create sale order.
+            - selected quants company is not same to user company.
+        """
+        context = dict(self._context or {})
+        active_ids = context.get('active_ids', [])
+        active_model = context.get('active_model')
+        quant_ids = self.env[active_model].browse(active_ids)
+        source_location = quant_ids[0].location_id
+        if any(q.location_id != source_location for q in quant_ids):
+            raise UserError(_('Please select quants that are in the same '
+                              'Location.'))
+        order_line_list = self.env['sale.order.line'].search([
+            ('state', 'not in', ['done', 'cancel']),
+            ('product_id', 'in', [q.product_id.id for q in quant_ids])
+        ])
+        if order_line_list:
+            error_msg = ''
+            for order_line in order_line_list:
+                error_msg += '\n%s: %s' % (
+                    order_line.order_id.name,
+                    order_line.product_id.display_name,
+                )
+            raise UserError(_('Some of the selected quants are already '
+                              'create sale order:%s') % error_msg)
+        if any(self .env.user.company_id.id != q.company_id.id for q in
+               quant_ids):
+            raise UserError(_('You cannot create sales order from stock '
+                              'quants that belongs to other company.'))
+        return super(QuantSaleOrderWizard, self).default_get(fields)
+
+    @api.multi
+    def action_create_sale_order(self):
+        '''This method create a sale order.'''
+        context = dict(self._context or {})
+        active_ids = context.get('active_ids', [])
+        stock_quant_obj = self.env['stock.quant']
+        quant_ids = stock_quant_obj.browse(active_ids)
+        warehouse_id = self.get_warehouse_id(quant_ids[0].location_id)
+        sale_order_obj = self.env['sale.order']
+        sale_order_line_obj = self.env['sale.order.line']
+        origin_name = ','.join([q.display_name for q in quant_ids])
+        order_vals = {
+            'partner_id': self.partner_id.id,
+            'team_id': self.team_id.id,
+            'company_id': self.env.user.company_id.id,
+            'state': 'draft',
+            'warehouse_id': warehouse_id.id,
+            'origin': origin_name,
```

Comment:
@yostashiro Can you review this part and let me know if there is any fields we need to add when creating the sales order

Reply:
I can't think of anything to add at the moment.  I think, however, that origin_name will be too long and I don't think it will carry any meaningful info.  We may delete this field unless you have a different idea.

Reply:
I agree the origin could be too long and does not carry much useful information. I will delete it